File: examples/example_plot_cross_sections.py
"""Plot example cross sections for the three ionization processes."""

from pathlib import Path

import matplotlib.pyplot as plt
import numpy as np

from py_ionization.charge_exchange import charge_exchange_cross_section
from py_ionization.electron_impact import electron_impact_cross_section
from py_ionization.photoionization import photoionization_cross_section


out_dir = Path("example_outputs")
out_dir.mkdir(exist_ok=True)

energy = np.logspace(1, 4, 300)
wavelength = np.linspace(1.0, 120.0, 300)

fig, axes = plt.subplots(1, 3, figsize=(10, 3.2), constrained_layout=True)

axes[0].plot(wavelength, photoionization_cross_section(wavelength, "H"))
axes[0].set_title("Photoionization, H")
axes[0].set_xlabel("Wavelength (nm)")
axes[0].set_ylabel("Cross section (cm$^2$)")

axes[1].plot(energy, electron_impact_cross_section(energy, "H"))
axes[1].set_title("Electron impact, H")
axes[1].set_xlabel("Energy (eV)")
axes[1].set_xscale("log")
axes[1].set_yscale("log")

axes[2].plot(energy, charge_exchange_cross_section(energy, ion="H+", neutral="H"))
axes[2].set_title("Charge exchange, H+ + H")
axes[2].set_xlabel("Energy (eV)")
axes[2].set_xscale("log")
axes[2].set_yscale("log")

fig.savefig(out_dir / "cross_section_examples.png", dpi=200)
print(f"Saved {out_dir / 'cross_section_examples.png'}")

File: matlab/examples/example_plot_cross_sections.m
% Plot cross sections from the MATLAB py_ionization interface.

clear; close all;

this_file = mfilename("fullpath");
examples_dir = fileparts(this_file);
matlab_dir = fileparts(examples_dir);
addpath(genpath(matlab_dir));

set(groot, "defaultAxesFontName", "Arial");
set(groot, "defaultTextFontName", "Arial");
set(groot, "defaultLegendFontName", "Arial");

wavelength_nm = linspace(0.1, 120, 300);
energy_eV = logspace(1, 3.5, 300);

ph_sigma = py_ionization.photoionization.cross_section(wavelength_nm, "H");
ei_sigma = py_ionization.electron_impact.cross_section(energy_eV, "H");
cex_sigma = py_ionization.charge_exchange.cross_section( ...
    energy_eV, "Ion", "H+", "Neutral", "H");

figure("Color", "w", "Position", [100, 100, 1200, 360]);

subplot(1, 3, 1);
plot(wavelength_nm, ph_sigma, "LineWidth", 2);
box on; grid on;
xlabel("Wavelength (nm)");
ylabel("Cross section (cm^2)");
title("Photoionization");

subplot(1, 3, 2);
loglog(energy_eV, ei_sigma, "LineWidth", 2);
box on; grid on;
xlabel("Energy (eV)");
ylabel("Cross section (cm^2)");
title("Electron impact");

subplot(1, 3, 3);
loglog(energy_eV, cex_sigma, "LineWidth", 2);
box on; grid on;
xlabel("Energy (eV)");
ylabel("Cross section (cm^2)");
title("Charge exchange");

out_dir = fullfile(fileparts(matlab_dir), "matlab", "comparison_outputs");
if ~isfolder(out_dir)
    mkdir(out_dir);
end
exportgraphics(gcf, fullfile(out_dir, "matlab_cross_sections.png"), "Resolution", 300);

